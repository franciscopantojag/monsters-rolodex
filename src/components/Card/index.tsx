import { Monster } from 'src/types';
import './styles.css';

interface Props {
  monster: Monster;
}

const Card: React.FC<Props> = ({ monster: { id, name, email } }) => (
  <div className="card-container" key={id}>
    <img src={`https://robohash.org/${id}?set=set2&size=180x180`} alt={`monster ${name}`} />
    <h1>{name}</h1>
    <p>{email}</p>
  </div>
);

export default Card;
