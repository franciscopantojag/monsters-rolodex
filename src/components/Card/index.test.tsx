import { shallow } from 'enzyme';
import Card from '.';

it('expect to render Card Component', () => {
  expect(shallow(<Card monster={{ id: 1, name: 'Hola', email: 'Hola' }} />)).toMatchSnapshot();
});
