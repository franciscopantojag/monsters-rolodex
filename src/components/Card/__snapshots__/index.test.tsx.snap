// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card Component 1`] = `
<div
  className="card-container"
  key="1"
>
  <img
    alt="monster Hola"
    src="https://robohash.org/1?set=set2&size=180x180"
  />
  <h1>
    Hola
  </h1>
  <p>
    Hola
  </p>
</div>
`;
